charts/sn-platform/templates/prometheus/_prometheus.tpl

```diff
@@ -130,6 +131,96 @@ ad.datadoghq.com/{{ template "pulsar.fullname" . }}-{{ .Values.prometheus.compon
     }
   ]
 {{- end }}
+{{- if eq .Values.datadog.adVersion "v2" }}
+ad.datadoghq.com/{{ template "pulsar.fullname" . }}-{{ .Values.prometheus.component }}.checks: |
+  {
+    "openmetrics": {
+      "init_config": [{}],
+      "instances": [
+    {
+      "prometheus_url": "http://%%host%%:{{ .Values.prometheus.port }}/federate?match[]=%7B__name__%3D~%22pulsar_.%2B%7Cjvm_.%2B%7Ctopic_.%2B%22%7D",
+      {{ if .Values.datadog.namespace -}}
+      "namespace": "{{ .Values.datadog.namespace }}",
+      {{ else -}}
+      "namespace": "{{ template "pulsar.namespace" . }}",
+      {{ end -}}
+      "metrics": {{ .Values.datadog.components.prometheus.metrics }},
+      "health_service_check": true,
+      "prometheus_timeout": 1000,
+      "max_returned_metrics": 1000000,
+      "type_overrides": {
+        "pulsar_topics_count": "gauge",
+        "pulsar_rate_in": "gauge",
+        "pulsar_rate_out": "gauge",
+        "pulsar_subscriptions_count": "gauge",
+        "pulsar_producers_count": "gauge",
+        "pulsar_consumers_count": "gauge",
+        "pulsar_throughput_in": "gauge",
+        "pulsar_throughput_out": "gauge",
+        "pulsar_storage_size": "gauge",
+        "pulsar_msg_backlog": "gauge",
+        "pulsar_storage_backlog_size": "gauge",
+        "pulsar_storage_offloaded_size": "gauge",
+        "pulsar_storage_write_latency_le_0_5": "gauge",
+        "pulsar_storage_write_latency_le_1": "gauge",
+        "pulsar_storage_write_latency_le_5": "gauge",
+        "pulsar_storage_write_latency_le_10": "gauge",
+        "pulsar_storage_write_latency_le_20": "gauge",
+        "pulsar_storage_write_latency_le_50": "gauge",
+        "pulsar_storage_write_latency_le_100": "gauge",
+        "pulsar_storage_write_latency_le_200": "gauge",
+        "pulsar_storage_write_latency_le_1000": "gauge",
+        "pulsar_storage_write_latency_overflow": "gauge",
+        "pulsar_entry_size_le_128": "gauge",
+        "pulsar_entry_size_le_512": "gauge",
+        "pulsar_entry_size_le_1_kb": "gauge",
+        "pulsar_entry_size_le_2_kb": "gauge",
+        "pulsar_entry_size_le_4_kb": "gauge",
+        "pulsar_entry_size_le_16_kb": "gauge",
+        "pulsar_entry_size_le_100_kb": "gauge",
+        "pulsar_entry_size_le_1_mb": "gauge",
+        "pulsar_entry_size_le_overflow": "gauge",
+        "pulsar_subscription_back_log": "gauge",
+        "pulsar_subscription_back_log_no_delayed": "gauge",
+        "pulsar_subscription_delayed": "gauge",
+        "pulsar_subscription_msg_rate_redeliver": "gauge",
+        "pulsar_subscription_unacked_messages": "gauge",
+        "pulsar_subscription_blocked_on_unacked_messages": "gauge",
+        "pulsar_subscription_msg_rate_out": "gauge",
+        "pulsar_subscription_msg_throughput_out": "gauge",
+        "pulsar_in_bytes_total": "counter",
+        "pulsar_in_messages_total": "counter",
```

Comment:
>  Note: To collect counter metrics with names ending in `_total`, specify the metric name without the `_total`
 suffix. For example, to collect the counter metric `promhttp_metric_handler_requests_total`, specify
 `promhttp_metric_handler_requests`. This submits to Datadog the metric name appended with `.count`.

Reply:
fixed in https://github.com/streamnative/charts/pull/1199/commits/76359e323aa89fa5bd1d10ada05ea3a01be351ee